Index: project/src/main/java/com/example/project/controller/document/VkrController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/src/main/java/com/example/project/controller/document/VkrController.java	(revision f537f66336350f7f5daceee1c8b9c5f09dfc74d6)
+++ project/src/main/java/com/example/project/controller/document/VkrController.java	(revision 54f3099f4b631913b907e7e2687b137fa16fd7ba)
@@ -20,6 +20,7 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Controller;
 
+import java.util.LinkedList;
 import java.util.List;
 
 @Controller
@@ -51,9 +52,6 @@
     @FXML
     private TableColumn<String, VkrDto> year ;
 
-
-
-
     @FXML
     private TableView<VkrDto> table;
 
@@ -66,6 +64,24 @@
     @FXML
     private TextField rukField;
 
+    @FXML
+    private TextField numberField;
+
+    @FXML
+    private TextField groupField;
+
+    @FXML
+    private TextField fioField;
+
+    @FXML
+    private TextField temaField;
+
+    @FXML
+    private TextField yearField;
+
+    @FXML
+    private TextField naprField;
+
     private List<VkrDto> vkrDtoList;
 
     @Qualifier("DocumentView")
@@ -112,6 +128,66 @@
         List<VkrDto> vkrDtos = vkrService.getAllByRuk(ruk);
         if(!vkrDtos.isEmpty()) {
             setTable(vkrDtos);
+        } else {
+
+        }
+    }
+
+    @FXML
+    private void clickFindNumber(ActionEvent event) {
+        String number = numberField.getText();
+        VkrDto vkrDto = vkrService.getVkrByNumber(number);
+        if(vkrDto != null) {
+            List<VkrDto> vkrDtoList = new LinkedList<>();
+            vkrDtoList.add(vkrDto);
+            setTable(vkrDtoList);
+        } else {
+
+        }
+    }
+
+    @FXML
+    private void clickFindGroup(ActionEvent event) {
+        String group = groupField.getText();
+        List<VkrDto> vkrDtoList = vkrService.getAllByGroup(group);
+        if(!vkrDtoList.isEmpty()) {
+            setTable(vkrDtoList);
+        }
+    }
+
+    @FXML
+    private void clickFindFio(ActionEvent event) {
+        String fio = fioField.getText();
+        List<VkrDto> vkrDtoList = vkrService.getAllByFio(fio);
+        if (!vkrDtoList.isEmpty()) {
+            setTable(vkrDtoList);
+        }
+    }
+
+    @FXML
+    private void clickFindTema(ActionEvent event) {
+        String tema = temaField.getText();
+        List<VkrDto> vkrDtoList = vkrService.getAllByTema(tema);
+        if(!vkrDtoList.isEmpty()) {
+            setTable(vkrDtoList);
+        }
+    }
+
+    @FXML
+    private void clickFindYear(ActionEvent event) {
+        String year = yearField.getText();
+        List<VkrDto> vkrDtoList = vkrService.getAllByYear(year);
+        if(!vkrDtoList.isEmpty()) {
+            setTable(vkrDtoList);
+        }
+    }
+
+    @FXML
+    private void clickFindNapr(ActionEvent actionEvent) {
+        String napr = naprField.getText();
+        List<VkrDto> vkrDtoList = vkrService.getAllByNapr(napr);
+        if(!vkrDtoList.isEmpty()) {
+            setTable(vkrDtoList);
         }
     }
 }
Index: project/src/main/java/com/example/project/repository/VkrRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/src/main/java/com/example/project/repository/VkrRepository.java	(revision f537f66336350f7f5daceee1c8b9c5f09dfc74d6)
+++ project/src/main/java/com/example/project/repository/VkrRepository.java	(revision 54f3099f4b631913b907e7e2687b137fa16fd7ba)
@@ -19,5 +19,17 @@
     List<Vkr> findAll();
 
     List<Vkr> findAllByRuk(Employee employee);
+
+    Vkr findByNumber(String number);
+
+    List<Vkr> findAllByGroup(String group);
+
+    List<Vkr> findAllByFio(String fio);
+
+    List<Vkr> findAllByTema(String tema);
+
+    List<Vkr> findAllByYear(String year);
+
+    List<Vkr> findAllByNapravlenie(String napr);
 }
 
Index: project/src/main/java/com/example/project/servise/VkrService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/src/main/java/com/example/project/servise/VkrService.java	(revision f537f66336350f7f5daceee1c8b9c5f09dfc74d6)
+++ project/src/main/java/com/example/project/servise/VkrService.java	(revision 54f3099f4b631913b907e7e2687b137fa16fd7ba)
@@ -10,4 +10,16 @@
 
     List<VkrDto> getAllByRuk(String rukFio);
 
+    VkrDto getVkrByNumber(String number);
+
+    List<VkrDto> getAllByGroup(String group);
+
+    List<VkrDto> getAllByFio(String fio);
+
+    List<VkrDto> getAllByTema(String tema);
+
+    List<VkrDto> getAllByYear(String year);
+
+    List<VkrDto> getAllByNapr(String napr);
+
 }
Index: project/src/main/java/com/example/project/servise/VkrServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/src/main/java/com/example/project/servise/VkrServiceImpl.java	(revision f537f66336350f7f5daceee1c8b9c5f09dfc74d6)
+++ project/src/main/java/com/example/project/servise/VkrServiceImpl.java	(revision 54f3099f4b631913b907e7e2687b137fa16fd7ba)
@@ -37,4 +37,40 @@
         List<Vkr> vkrList = vkrRepository.findAllByRuk(employee);
         return vkrMapper.convertModelsToDtos(vkrList);
     }
+
+    @Override
+    public VkrDto getVkrByNumber(String number) {
+        Vkr vkr = vkrRepository.findByNumber(number);
+        return vkrMapper.convertModelToDto(vkr);
+    }
+
+    @Override
+    public List<VkrDto> getAllByGroup(String group) {
+        List<Vkr> vkrList = vkrRepository.findAllByGroup(group);
+        return vkrMapper.convertModelsToDtos(vkrList);
+    }
+
+    @Override
+    public List<VkrDto> getAllByFio(String fio) {
+        List<Vkr> vkrList = vkrRepository.findAllByFio(fio);
+        return vkrMapper.convertModelsToDtos(vkrList);
+    }
+
+    @Override
+    public List<VkrDto> getAllByTema(String tema) {
+        List<Vkr> vkrList = vkrRepository.findAllByTema(tema);
+        return vkrMapper.convertModelsToDtos(vkrList);
+    }
+
+    @Override
+    public List<VkrDto> getAllByYear(String year) {
+        List<Vkr> vkrList = vkrRepository.findAllByYear(year);
+        return vkrMapper.convertModelsToDtos(vkrList);
+    }
+
+    @Override
+    public List<VkrDto> getAllByNapr(String napr) {
+        List<Vkr> vkrList = vkrRepository.findAllByNapravlenie(napr);
+        return vkrMapper.convertModelsToDtos(vkrList);
+    }
 }
Index: project/src/main/resources/fxml/VKR.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- project/src/main/resources/fxml/VKR.fxml	(revision f537f66336350f7f5daceee1c8b9c5f09dfc74d6)
+++ project/src/main/resources/fxml/VKR.fxml	(revision 54f3099f4b631913b907e7e2687b137fa16fd7ba)
@@ -24,19 +24,19 @@
       <Label layoutX="447.0" layoutY="4.0" prefHeight="45.0" prefWidth="356.0" text="Темы выпускных квалификационных работ магистров" />
       <Button fx:id="loadButton" layoutX="1009.0" layoutY="14.0" mnemonicParsing="false" onAction="#clickLoadButton" prefHeight="25.0" prefWidth="109.0" text="Загрузить" />
       <Button fx:id="addButton" layoutX="69.0" layoutY="14.0" mnemonicParsing="false" onAction="#clickLoadButton" prefHeight="25.0" prefWidth="152.0" text="Добавить данные" />
-      <Button fx:id="numberButton" layoutX="49.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="numberButton" onAction="#clickFindNumber" layoutX="49.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
       <TextField fx:id="numberField" layoutX="7.0" layoutY="49.0" prefHeight="25.0" prefWidth="42.0" />
-      <Button fx:id="groupButton" layoutX="155.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
-      <TextField fx:id="groupField" layoutX="79.0" layoutY="49.0" prefHeight="25.0" prefWidth="76.0" />
+      <Button fx:id="groupButton" onAction="#clickFindGroup" layoutX="155.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <TextField fx:id="groupField"  layoutX="79.0" layoutY="49.0" prefHeight="25.0" prefWidth="76.0" />
       <TextField fx:id="naprField" layoutX="183.0" layoutY="49.0" prefHeight="25.0" prefWidth="199.0" />
-      <Button fx:id="naprButton" layoutX="382.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="naprButton" onAction="#clickFindNapr" layoutX="382.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
       <TextField fx:id="fioField" layoutX="415.0" layoutY="49.0" prefHeight="25.0" prefWidth="130.0" />
-      <Button fx:id="fioButton" layoutX="545.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="fioButton" onAction="#clickFindFio" layoutX="545.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
       <TextField fx:id="temaField" layoutX="573.0" layoutY="49.0" prefHeight="25.0" prefWidth="319.0" />
-      <Button fx:id="temaButton" layoutX="892.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="temaButton" onAction="#clickFindTema" layoutX="892.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
       <TextField fx:id="rukField" layoutX="924.0" layoutY="49.0" prefHeight="25.0" prefWidth="174.0" />
-      <Button fx:id="rukButton" layoutX="1098.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="rukButton" onAction="#clickFindRuk" layoutX="1098.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
       <TextField fx:id="yearField" layoutX="1130.0" layoutY="49.0" prefHeight="25.0" prefWidth="93.0" />
-      <Button fx:id="yearButton" layoutX="1223.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
+      <Button fx:id="yearButton" onAction="#clickFindYear" layoutX="1223.0" layoutY="49.0" mnemonicParsing="false" prefHeight="25.0" prefWidth="20.0" />
    </children>
 </Pane>
